contracts/pangolin-roll/PangolinRoll.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
```

Comment:
we can certainly use solidity 0.8.0+ and interact with other contracts that are earlier versions. hit me up and we can work through how dependencies for both versions can be managed in the repo